src/command_modules/azure-cli-ams/azure/cli/command_modules/ams/_params.py

```diff
@@ -0,0 +1,151 @@
+    password_arg_type = CLIArgumentType(options_list=('--password', '-p'), metavar='PASSWORD_NAME')
+    transform_name_arg_type = CLIArgumentType(options_list=('--transform-name', '-t'), metavar='TRANSFORM_NAME')
+    expiry_arg_type = CLIArgumentType(options_list=('--expiry'), metavar='EXPIRY_TIME')
+
+    with self.argument_context('ams account') as c:
+        c.argument('account_name', name_arg_type, id_part='name',
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('location', arg_type=get_location_type(self.cli_ctx),
+                   validator=get_default_location_from_resource_group)
+        c.argument('tags', arg_type=tags_type)
+
+    with self.argument_context('ams account create') as c:
+        c.argument('storage_account', storage_account_arg_type,
+                   help='The name or resource ID of the primary storage account to attach to the Azure Media Services account. Blob only accounts are not allowed as primary.',
+                   validator=storage_account_id)
+
+    with self.argument_context('ams account storage') as c:
+        c.argument('account_name', account_name_arg_type,
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('storage_account', name_arg_type,
+                   help='The name or resource ID of the secondary storage account to detach from the Azure Media Services account.',
+                   validator=storage_account_id)
+
+    with self.argument_context('ams account sp') as c:
+        c.argument('account_name', account_name_arg_type,
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('sp_name', name_arg_type,
+                   help="The app name or app URI to associate the RBAC with. If not present, a default name like '{amsaccountname}-access-sp' will be generated.")
+        c.argument('sp_password', password_arg_type,
+                   help="The password used to log in. Also known as 'Client Secret'. If not present, a random secret will be generated.")
+        c.argument('role', completer=get_role_definition_name_completion_list)
+        c.argument('xml', help='Enables xml output format.')
+        c.argument('years', type=int, default=None)
+
+    with self.argument_context('ams transform') as c:
+        c.argument('account_name', account_name_arg_type, id_part='name',
+                   help='The name of the Azure Media Services account within the resource group.')
+        c.argument('transform_name', name_arg_type, id_part='child_name_1',
+                   help='The name of the transform.')
+        c.argument('preset_names', arg_type=get_enum_type(EncoderNamedPreset),
+                   nargs='+',
+                   help='Space-separated list of built preset names.')
+        c.argument('description', help='Customer supplied description of the transform.')
```

Comment:
"Customer-supplied" though this could also just be removed (all CLI values are customer-supplied ^_^)

Reply:
Done